### PR TITLE
SOL-56505

### PR DESCRIPTION
**Pull request by PhilippeKhalife**
Contribute code snippets.

### CODE REVIEW

File: src/main/java/com/solace/samples/jms/snippets/HowToCreateAConnectionForOAuth.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Demonstrates how to create a Connection configured for OAuth2.0 
+ */
+public class HowToCreateAConnectionForOAuth {
+
+  final static String SOLJMS_INITIAL_CONTEXT_FACTORY = "com.solacesystems.jndi.SolJNDIInitialContextFactory";
+  final static String CONNECTION_FACTORY_JNDI_NAME = "/JNDI/CF/GettingStarted";
+
+  public static Connection createJmsConnectionWithOauth(String jndiProviderURL,String myAccessToken,String issuerIdentifiereURL) throws NamingException,JMSException
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jms/snippets/HowToCreateAConnectionForOAuth.java]: unless this is Canadian English I think `issuerIdentifiereURL` should be `issuerIdentifierURL` ?

File: src/main/java/com/solace/samples/jms/snippets/HowToCreateAConnectionForOAuth.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  {
+    Hashtable<String, Object> env = new Hashtable<String, Object>();
+    env.put(InitialContext.INITIAL_CONTEXT_FACTORY, SOLJMS_INITIAL_CONTEXT_FACTORY);
+    env.put(InitialContext.PROVIDER_URL, jndiProviderURL);
+    env.put(SupportedProperty.SOLACE_JMS_AUTHENTICATION_SCHEME, SupportedProperty.AUTHENTICATION_SCHEME_OAUTH2);
+    env.put(SupportedProperty.SOLACE_JMS_OAUTH2_ACCESS_TOKEN, myAccessToken);
+    env.put(SupportedProperty.SOLACE_JMS_OAUTH2_ISSUER_IDENTIFIER, issuerIdentifiereURL);
```
Comment (Mrc0113) [inline src/main/java/com/solace/samples/jms/snippets/HowToCreateAConnectionForOAuth.java]: same comment as above

**Review by Mrc0113 — CHANGES_REQUESTED** [review]
Thanks @PhilippeKhalife. Looks good besides that 1 typo

File: src/main/java/com/solace/samples/jms/snippets/HowToCreateAConnectionForOAuth.java (hunk 1 above)
Comment (PhilippeKhalife) [reply src/main/java/com/solace/samples/jms/snippets/HowToCreateAConnectionForOAuth.java]: I should have checked that from the source, we can pretend it is Canadian with some french :) - I will fix it !!

**Review by Mrc0113 — APPROVED** [review]

**Pull request merged**